src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -739,6 +740,108 @@ class StringGenerator : public Generator<std::string> {
     const size_t _alphabetLength;
 };
 
+
+class DataSetCache {
+public:
+    DataSetCache() {}
+
+    /* The dataset is stored in a map, using the path of the file as the key and a vector of strings
+    with file's content as the value. In this method the empty vector is created and return its
+    pointer */
```

Comment:
For the docs, one suggestion is a note on this class that says the getDataSetForPath method doesn't need the lock because loading and evaluating are separate stages in genny.

Reply:
This is done.